proj_eclipse/src/main/java/smartmag/models/OrderModel.java

```diff
@@ -0,0 +1,370 @@
+				order);
+		if (!instances.containsKey(order))
+			instances.put(order.getId(), this); // carico OrderModel in hashmap
+	}
+
+	/**
+	 * Crea un nuovo record dell'ordine usando l'ordine del modello
+	 */
+	public void createOrdineRecord()
+			throws SQLIntegrityConstraintViolationException, ParseException {
+
+		if (orderIsSavedInDb())
+			throw new SQLIntegrityConstraintViolationException(
+					"Ordine #" + ordine.getId() + " esiste già!");
+
+		OrdineRecord or = DSL.newRecord(ORDINE);
+		or.setId(ordine.getId());
+		copyOrdineIntoRecord(ordine, or);
+		or.store(); // INSERT
+		this.orderRecord = or;
+
+		// TODO: event
+	}
+
+	/**
+	 * Cerca se esiste il record dell'ordine, se esiste lo aggiorna utilizzando
+	 * i valori dell'ordine passato al metodo
+	 * 
+	 * @param o ordine del quale si vuole aggiornare il record
+	 */
+	public void updateOrdine(Ordine o)
+			throws SQLIntegrityConstraintViolationException, ParseException {
+		if (!orderIsSavedInDb())
+			throw new SQLIntegrityConstraintViolationException(
+					"Ordine #" + o.getId() + " non esiste!");
+
+		orderRecord = (OrdineRecord) fetchOrderRecordById(o.getId());
+		copyOrdineIntoRecord(o, orderRecord);
+		orderRecord.store(); // UPDATE con UpdatableRecord
+	}
+
+	/**
+	 * Aggiorna lo stato dell'ordine sia nel record che nel parametro ordine che
+	 * è stato passato al metodo
+	 * 
+	 * @param o        ordine del quale si vuole modificare lo stato
+	 * @param newStato nuovo stato dell'ordine
+	 * @throws ParseException se newStato != ENUM
+	 */
+	public void updateStatoOrdine(Ordine o, String newStato)
+			throws SQLIntegrityConstraintViolationException, ParseException {
+		if (!orderIsSavedInDb())
+			throw new SQLIntegrityConstraintViolationException(
+					"Ordine #" + o.getId() + " non esiste!");
+
+		o.setStato(StatoOrdine.valueOf(newStato));
+		updateOrdine(o);
+	}
+
+	/**
+	 * cancella il record dell'ordine nel modello
+	 */
+	public void deleteOrdine() throws ParseException {
+		if (orderIsSavedInDb()) {
+			orderRecord.delete(); // DELETE con UpdatableRecord
+		}
+	}
+
+	/**
+	 * Aggiorna l'ordine prendendo i dati dal record di quell'ordine
+	 */
+	// aggiunte eccezioni a causa delle conversioni
+	private void refreshOrderFromDb() throws ParseException {
+		this.orderRecord = fetchOrderRecordById(ordine.getId());
+		if (this.orderRecord != null) {
+			Ordine o = ordineFromRecord(this.orderRecord);
+			this.ordine.setTipo(o.getTipo());
+			this.ordine.setStato(o.getStato());
+			this.ordine.setDataEmissione(o.getDataEmissione());
+			this.ordine.setDataCompletamento(o.getDataCompletamento());
+		}
+		// TODO: event
+	}
+
+	/**
+	 * Controlla che ci sia un record di quell'ordine nel DB
+	 * 
+	 * @return ritorna 1 se è presente e 0 se non esiste
+	 */
+	public boolean orderIsSavedInDb() throws ParseException {
+		refreshOrderFromDb();
+		return orderRecord != null;
+	}
+
+	/**
+	 * crea un nuovo record di prodottoOrdine
+	 */
+	public void createProdottoOrdineRecord() {
```

Comment:
anche questo private
```suggestion
	private void createProdottoOrdineRecord() {
```